src/settings/MatchProfiles/MatchProfilesForm.js

```diff
@@ -1,49 +1,80 @@
-import React, { useState } from 'react';
+import React, {
+  memo,
+  useState,
+} from 'react';
 import PropTypes from 'prop-types';
 import queryString from 'query-string';
 import { FormattedMessage } from 'react-intl';
 import { connect } from 'react-redux';
-import { get } from 'lodash';
+import {
+  get,
+  isEmpty,
+} from 'lodash';
 
 import stripesForm from '@folio/stripes/form';
 
 import {
   FlexibleForm,
   FOLIO_RECORD_TYPES,
 } from '../../components';
-import { compose } from '../../utils';
+import {
+  compose,
+  matchFields,
+  getDropdownOptions,
+} from '../../utils';
 import { LAYER_TYPES } from '../../utils/constants';
 import { formConfigSamples } from '../../../test/bigtest/mocks';
 
 import styles from './MatchProfilesForm.css';
 
 const formName = 'matchProfilesForm';
 
-export const MatchProfilesFormComponent = ({
+export const MatchProfilesFormComponent = memo(({
   pristine,
   submitting,
-  initialValues,
+  initialValues: {
+    existingRecordType,
+    name,
+    matchDetails,
+  },
   handleSubmit,
   location: { search },
   associatedJobProfilesAmount,
   onCancel,
+  jsonSchemas,
 }) => {
-  const [isConfirmEditModalOpen, setConfirmModalOpen] = useState(false);
-
   const { layer } = queryString.parse(search);
   const isEditMode = layer === LAYER_TYPES.EDIT;
-  const isSubmitDisabled = pristine || submitting;
 
-  const paneTitle = isEditMode ? (
-    <FormattedMessage id="ui-data-import.edit">
-      {txt => `${txt} ${initialValues.name}`}
-    </FormattedMessage>
-  ) : <FormattedMessage id="ui-data-import.settings.matchProfiles.new" />;
-  const headLine = isEditMode ? initialValues.name : <FormattedMessage id="ui-data-import.settings.matchProfiles.new" />;
+  const [selectedExistingRecord, setSelectedExistingRecord] = useState(isEditMode ? existingRecordType : '');
+  const [existingRecordFields, setExistingRecordFields] = useState([]);
+  const [isConfirmEditModalOpen, setConfirmModalOpen] = useState(false);
+
+  const isSubmitDisabled = pristine || submitting;
+  const paneTitle = isEditMode
+    ? (
+      <FormattedMessage id="ui-data-import.edit">
+        {txt => `${txt} ${name}`}
+      </FormattedMessage>
+    )
+    : <FormattedMessage id="ui-data-import.settings.matchProfiles.new" />;
+  const headLine = isEditMode
+    ? name
+    : <FormattedMessage id="ui-data-import.settings.matchProfiles.new" />;
```

Comment:
The same `FormatterMessage` used in string 60, maybe it's a reasone to make a variable which would contain it. 

Or, `paneTitle` and `headline` have indentical conditions, it could be more readable if you cover it in a function, for example:

`const newMatchProfileLabel = <FormattedMessage id="ui-data-import.settings.matchProfiles.new" />;`
`const getLabelForEditMode = (value) => isEditMode ? value : newMatchProfileLabel;`